contrib/src/main/java/macrobase/analysis/transform/aggregate/BatchWindowAvg.java

```diff
@@ -0,0 +1,39 @@
+package macrobase.analysis.transform.aggregate;
+
+import macrobase.conf.ConfigurationException;
+import macrobase.conf.MacroBaseConf;
+import macrobase.conf.MacroBaseDefaults;
+import macrobase.datamodel.Datum;
+import org.apache.commons.math3.linear.ArrayRealVector;
+import org.apache.commons.math3.linear.RealVector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BatchWindowAvg extends BatchWindowAggregate {
+    public BatchWindowAvg(){}
+
+    public BatchWindowAvg(MacroBaseConf conf) throws ConfigurationException {
+        this.timeColumn = conf.getInt(MacroBaseConf.TIME_COLUMN, MacroBaseDefaults.TIME_COLUMN);
+    }
+
+    public Datum aggregate(List<Datum> data) {
+        if (data.isEmpty())
+            return new Datum(new ArrayList<>(), new ArrayRealVector(dim));
+
+        dim = data.get(0).metrics().getDimension();
+        RealVector results = new ArrayRealVector(dim);
```

Comment:
Yes -- ASAP is going to use BatchWindowAvg as the transform. Right now we only have BatchWindowMax.
And sounds good. I will switch to using double[].